src/script/interpreter.cpp

```diff
@@ -1230,6 +1230,17 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                 }
                 break;
 
+                case OP_INTERNALKEY: {
+                    // OP_INTERNALKEY is only available in Tapscript
+                    if (sigversion == SigVersion::BASE || sigversion == SigVersion::WITNESS_V0) {
+                        return set_error(serror, SCRIPT_ERR_BAD_OPCODE);
+                    }
+                    // Always present in Tapscript
+                    assert(execdata.m_internal_key);
```

Comment:
`assert(sigversion == TAPSCRIPT)` in that case?

Reply:
added